transport/properties.go

```diff
@@ -0,0 +1,59 @@
+package transport
+
+// Properties provides storing and reading metadata values. Keys may be any
+// comparable value type. Get and set will panic if key is not a comparable
+// value type.
+//
+// Properties uses lazy initialization, and Set method must be called as an
+// addressable value, or pointer. Not doing so may cause key/value pair to not
+// be set.
+type Properties struct {
+	values map[interface{}]interface{}
+}
```

Comment:
This file in `transport` isn't needed since the `Properties` are defined in the root package now?